go/vt/sqlparser/parse_test.go

```diff
@@ -667,6 +679,17 @@ func TestValid(t *testing.T) {
 	}, {
 		input:  "select k collate 'latin1_german2_ci' as k1 from t1 order by k1 asc",
 		output: "select k collate latin1_german2_ci as k1 from t1 order by k1 asc",
+	}, {
+		input:  "select /* drop trailing semicolon */ 1 from dual;",
+		output: "select /* drop trailing semicolon */ 1 from dual",
+	}, {
+		input: "select /* cache directive */ SQL_NO_CACHE 'foo' from t",
+	}, {
+		input: "select binary 'a' = 'A' from t",
+	}, {
+		input: "select match(a) against ('foo') from t",
+	}, {
+		input: "select match(a1, a2) against ('foo' in natural language mode with query expansion) from t",
```

Comment:
can you also add a test for the semi-colon? not seeing it here. anything else we're missing?